docs/source/tutorial/mutations.mdx

```diff
@@ -6,137 +6,132 @@ description: Learn how to update data with the useMutation hook
-
-To create a better experience for our users, we want to persist the login between sessions. In order to do that, we need to save our login token to `localStorage`. Let's learn how we can use the `onCompleted` handler of `useMutation` to persist our login:
+* The first object in `useMutation`'s result tuple (`login`) is the mutate function we call to execute the mutation. We pass this function to our `LoginForm` component.
+* The second object in the tuple is similar to the result object returned by `useQuery`, including fields for the operation's `loading` and `error` states and the operation's result `data`.
 
-### Expose Apollo Client with useApolloClient
+Now whenever a user submits the login form, our `login` mutation is called. The user's `token` is stored in the in-memory cache, _however_ we want that token to be available across multiple visits in the same browser. Let's tackle that next.
 
-Apollo Client ensures that your `ApolloClient` instance is available through React's context. Sometimes, we need to access the `ApolloClient` instance to directly call a method that isn't exposed by the `@apollo/client` helper components. The `useApolloClient` hook can help us access the client.
+#### Persist the user token
 
-Let's call `useApolloClient` to get the currently configured client instance. Next, we want to pass an `onCompleted` callback to `useMutation` that will be called once the mutation is complete with its return value. This callback is where we will save the login token to `localStorage`.
+In our call to `useMutation`, we can include an `onCompleted` callback. This enables us to interact with the mutation's result data as soon as it's available. We'll use this callback to persist the user's `token`.
 
-In our `onCompleted` handler, we also call `client.writeData` to write local data to the Apollo cache indicating that the user is logged in. This is an example of a **direct write** that we'll explore further in the next section on local state management.
+Modify the `useMutation` call in `login.tsx` to match the following:
 
-<MultiCodeBlock>
-
-```tsx{2-3,5-10,13-14}:title=src/pages/login.tsx
-export default function Login() {
-  const client: ApolloClient<any> = useApolloClient();
-  const [login, { loading, error }] = useMutation<LoginTypes.login, LoginTypes.loginVariables>(
-    LOGIN_USER,
-    {
-      onCompleted({ login }) {
-        localStorage.setItem('token', login as string);
-        client.writeData({ data: { isLoggedIn: true } });
-      }
+```tsx:title=src/pages/login.tsx
+const [login, { loading, error }] = useMutation<
+  LoginTypes.Login,
+  LoginTypes.LoginVariables
+>(
+  LOGIN_USER,
+  {
+    onCompleted({ login }) {
+      localStorage.setItem('token', login.token as string);
+      localStorage.setItem('userId', login.id as string);
```

Comment:
```suggestion
    onCompleted({ login: { token, id }}) {
      localStorage.setItem('token', token as string);
      localStorage.setItem('userId', id as string);
```
I'm not necessarily proposing this specific change, but I'm curious if you have any strong feelings about these different styles?

Reply:
No strong feelings, just matching what's in the tutorial source. Happy to tweak in both spots if preferable.